### ISSUE TITLE
Failed to run jobs.

### ISSUE THREAD

**Issue by cazorla19** [ansible-semaphore/semaphore#49]
Hi. I tried to run any job that has been created, but I did. Added a deploy key, the same that I synced my semaphore-host with Github, put an address of my repo, added group, which name is the same that in /etc/ansible/hosts, but job is failed.
Task output: Updating SSH Keys; SSH Keys Updated.
Checked this issue in Google, but nobody forced with it. Screenshots of my semaphore is below, if I didn`t explained clear.
![screenshot_1](https://cloud.githubusercontent.com/assets/14975946/10636320/d577b9a6-7807-11e5-97d0-10efd4efaa2e.png)
![screenshot_2](https://cloud.githubusercontent.com/assets/14975946/10636323/d57d2e68-7807-11e5-8f32-c1f0e86bbf94.png)
![screenshot_3](https://cloud.githubusercontent.com/assets/14975946/10636322/d57c14ce-7807-11e5-8689-1f570f7c1848.png)
![screenshot_4](https://cloud.githubusercontent.com/assets/14975946/10636321/d577bf64-7807-11e5-8dad-2bf607d8506f.png)

**Comment by matejkramny** [comment]
It could be waiting on something.

Can you explain what `/etc/ansible` file does?
As far as I know, semaphore assembles the inventory on its own

**Comment by matejkramny** [comment]
This runs all things ansible related: [runner.js](https://github.com/ansible-semaphore/semaphore/blob/master/lib/runner.js)

It spawns sub processes -- nothing special.

**Comment by cazorla19** [comment]
I read it, think there might be a problem with SSH keys checking. I`ve regenerated SSH key and put it in identity, but result is the same. Do you know occasionally, what can I do to solve it?

**Comment by matejkramny** [comment]
Hmm for starters i dont think you should be needing to modify anything aside from inside of semaphore.

You could try and manually ssh into the target host, it might be that its git waiting for an interaction to connect to an unknown ssh host (there is a wrapper used for it, but it might not be working in this case perhaps)

**Comment by cazorla19** [comment]
That`s the point, now I just have one host with Ansible and Semaphore and who should be target during the time, when semaphore runs. And SHH key on this host is the same that I put in semaphore identity.

**Comment by carlos-alarcon** [comment]
I was stuck at the same point (report only says it updated the keys) as you but went to runner.js and added some debug lines to the errors and found out that there were some missing files due to HOME env var not being defined.

My experience, installing semaphore in a ubuntu 14.04 VM (last weekend) which may be helpfull:
- Cloned semaphore under /opt
- Installed ansible 1.9.1 (followed the instructions in https://www.michaelrigart.be/en/blog/testing-ansible-roles-with-semaphore.html)
-  playbooks/playbook.yml locally 
-  chown -R semphore:semaphore /opt/semaphosre
- Edit /opt/semaphore/runit_semaphore/run so it has the HOME env set in the command line to /root

I had to hardcode HOME to /root  since the node app seemed to be run as root and I don't know why the process wasn't inheriting HOME var from shell 

It would be nice if we get a way of making the node app being run as semaphore user instead of root but due to my lack of knowledge  of 'runsv' I wasn't able.

**Comment by matejkramny** [comment]
@carlos-alarcon /root is currently hard-coded into the executor (runner.js file) as it was intended to be ran inside docker / environment where semaphore is the only thing. Obviously not well designed, but that will change in upcoming versions of the API

**Comment by carlos-alarcon** [comment]
@matejkramny seems that '/root' is not hardcoded in runner.js any more: 
`var home = process.env.HOME + '/';`
Anyway, in my machine, somehow the  /opt/semaphore/runit_semaphore/run wasn't inherint HOME enviroment variable (perhaps a 'srv' stuff, i never used 'srv' before so can't tell). So I edited /opt/sempahore/runit_semaphore/run to be like:

`#!/bin/bash

cd /opt/semaphore/
HOME=/root PORT=3000 node bin/semaphore.js`

which solved my problem (of not being able to run tasks).  I guess that if i manage this /etc/semaphore/runit_semaphore/run to be launched by 'semaphore' user then it would still work but using other folders (HOME=/home/semaphore) for the playbooks.

**Comment by matejkramny** [comment]
Hmm I don't know what `runit` is either, but from experience with process managers, they sometimes don't inherit outside environment to the process unless explicitly directed.

**Comment by gvenka008c** [comment]
@matejkramny Does semaphore download all the playbook from git and store it in some location when running as docker?

# docker ps
CONTAINER ID        IMAGE                    COMMAND                  CREATED             STATUS              PORTS                        NAMES
b29051cc8349        castawaylabs/semaphore   "node /srv/semaphore/"   45 hours ago        Up 45 hours         0.0.0.0:80->80/tcp           semaphore
ba4f55661d2c        mongo                    "/entrypoint.sh mongo"   45 hours ago        Up 45 hours         127.0.0.1:27017->27017/tcp   mongodb
7b0428cf86a8        redis                    "/entrypoint.sh redis"   46 hours ago        Up 46 hours         6379/tcp                     redisio

I am looking a way to find out how the playbook is getting called? will it be called from within docker container that's running semaphore? I was able to bring up the GUI and add ssh keys but for some reason the ssh to my remote machines where playbook needs to run are failing.

Please suggest.

**Comment by matejkramny** [comment]
It gets called when you run the job. Specifically look at `lib/runner.js`.

It downloads the playbook every time and then gets rid of it. Playbooks are downloaded to `$HOME/`

**Comment by gvenka008c** [comment]
@matejkramny Does semaphore downloads only the playbook or it will download my entires files / folders from the git repo? like all the roles, inventory files etc...

**Comment by matejkramny** [comment]
It will download the entire git repo, then tell ansible to use the playbook file

**Comment by tbynum** [comment]
I have a related question. I am trying to run semaphore on AWS in an EC2 instance. My github is behind the corporate firewall. I am having difficulty getting the SSH stuff set up. I have added the proxy info in the  run file but it is not even trying to hit the github repo. I would also like to find a way to use one set of SSH keys to connect to github and a different set to connect to the EC2 instances I am trying to manage. Any thoughts?

Tim

**Comment by matejkramny** [comment]
@tbynum please create a separate issue
